src/transformers/models/mask2former/modeling_mask2former.py

```diff
@@ -0,0 +1,3393 @@
+            encoder_hidden_states=tuple(features) if output_hidden_states else (),
+            decoder_hidden_states=decoder_output.hidden_states if output_hidden_states else (),
+            multi_scale_features=decoder_output.multi_scale_features,
+        )
+
+
+class Mask2FormerTransformerModule(nn.Module):
+    """
+    The Mask2Former's transformer module.
+    """
+
+    def __init__(self, in_features: int,  config: Mask2FormerConfig):
+        super().__init__()
+        #in_features -> in_channels
+        hidden_size = config.decoder_config.hidden_size #hidden_dim
+        print("hidden_size:",hidden_size)
+        print("in_features:",in_features)
+        should_project = in_features != hidden_size 
+        #self.pe_layer
+        self.position_embedder = Mask2FormerSinePositionEmbedding(num_pos_feats=hidden_size // 2, normalize=True)
+        #self.query_feat
+        self.learnable_queries = nn.Embedding(config.decoder_config.num_queries, hidden_size)
+        #self.query_embed
+        self.queries_embedder = nn.Embedding(config.decoder_config.num_queries, hidden_size)
+        
+        self.num_feature_levels = 3
+        self.level_embed = nn.Embedding(self.num_feature_levels, hidden_size)
+
+        self.input_projection = nn.ModuleList([
+            nn.Conv2d(in_features, hidden_size, kernel_size=1) 
+            if should_project else nn.Sequential() for _ in range(self.num_feature_levels)
+        ])
+        print("self.input_projection:",self.input_projection)
+        
+        self.decoder = DetrDecoder(config=config.decoder_config, mask_feature_size=config.mask_feature_size)
+
+
+    def forward(
+        self, 
+        multi_scale_features: torch.Tensor, 
+        pixel_embeddings: torch.Tensor,
+        output_hidden_states: bool = False, 
+        output_attentions: bool = False
+    ) -> DetrDecoderOutput:
+        input_embeds = []
+        position_embeddings = []
+        feature_size_list = []
+
+        for i in range(self.num_feature_levels):
+            feature_size_list.append(multi_scale_features[i].shape[-2:])
+            position_embeddings.append(self.position_embedder(multi_scale_features[i], None).flatten(2))
+            input_embeds.append(self.input_projection[i](multi_scale_features[i]).flatten(2) + self.level_embed.weight[i][None, :, None])
+            position_embeddings[-1] = position_embeddings[-1].permute(2,0,1)
+            input_embeds[-1] = input_embeds[-1].permute(2,0,1)
+
+        _, batch_size, _ = input_embeds[0].shape
+
+        queries_embeddings = self.queries_embedder.weight.unsqueeze(1).repeat(1, batch_size, 1)
+        learnable_queries = self.learnable_queries.weight.unsqueeze(1).repeat(1, batch_size, 1)
+
+        print("learnable_queries:",learnable_queries.shape)
+        print("queries_embeddings:",queries_embeddings.shape)
+        print("feature_size_list:",feature_size_list)
+
+        decoder_output: DetrDecoderOutput = self.decoder(
```

Comment:
No need to typecast a second time:
```suggestion
        output = self.decoder(
```

Reply:
oh yes right, thanks for pointing it out.